translate biblio types in preview and filter menu...

SteffenAstheimer opened the pull request:
... and adjust the lang:translate function to take optional variable data-template-keys.

SteffenAstheimer committed:
translate biblio types in preview and filter menu...

... and adjust the lang:translate function to take optional variable data-template-keys.

peterstadler reviewed (changes requested):
Thanks for the PR!
I was a little bit puzzled that this (translate by attribute) was not implemented yet?!
This is a valid use case but I would approach it a little bit differently, since the [templating module](https://exist-db.org/exist/apps/doc/templating) automatically scans for those "data-template-*" attributes and makes them available as parameters to functions.
So, my proposal would be to add a 3-arity version of the function like `lang:translate($node as node(), $model as map(*), $key as xs:string)`.

SteffenAstheimer committed:
[not yet working] Split lang:translate into 2- and 3-arity versions...

- 2-arity version translates the element's text content.
- 3-arity version translates the value of a key from the model, where the key is passed via @data-template-key.

SteffenAstheimer commented:
Thanks for the hint on the templating module documentation! That looks like a much cleaner solution. But for some reason the 3-arity version seems to constantly be shadowed by the 2-arity version. Is function overloading not working in the templating module? (I couln't find any mentioning of that in the documentation.)

I oploaded my current status. Maybe you'll know why the 3-arity function isn't called even if the template has a `@data-template-key='biblioType'` attribute.
I made sure the function itself is working flawlessly (to my best knowledge) by renaming it – but changed it back in my commit.

peterstadler committed:
rename function and clean up element constructor

Co-Authored-By: SteffenAstheimer <steffen.astheimer@uni-paderborn.de>
